nixos/modules/security/gnupg.nix

```diff
@@ -0,0 +1,270 @@
+{ config, lib, pkgs, ... }:
+let
+  inherit (builtins) attrNames dirOf length match split;
+  inherit (lib) types;
+  cfg = config.security.gnupg;
+  gnupgHome = "/var/lib/gnupg";
+  # Escape as required by: https://www.freedesktop.org/software/systemd/man/systemd.unit.html
+  escapeUnitName = name:
+    lib.concatMapStrings (s: if lib.isList s then "-" else s)
+    (split "[^a-zA-Z0-9_.\\-]+" name);
+in
+{
+options.security.gnupg = {
+  store = lib.mkOption {
+    type = with types; either str path;
+    default = "/root/.password-store";
+    description = ''
+      Default base path for the <literal>gpg</literal> option
+      of the <link linkend="opt-security.gnupg.secrets">secrets</link>.
+      Note that you may set it up with something like:
+      <literal>builtins.getEnv "PASSWORD_STORE_DIR" + "/machines/example"</literal>.
+    '';
+  };
+  secrets = lib.mkOption {
+    description = "Available secrets.";
+    default = {};
+    example = {
+      "/root/.ssh/id_ed25519" = {};
+      "knot/tsig/example.org/acme.conf" = {
+        user = "knot";
+      };
+      "lego/example.org/rfc2136" = {
+        pipe = "
+          cat -
+          cat <EOF
+          RFC2136_NAMESERVER=ns.example.org:53
+          RFC2136_TSIG_ALGORITHM=hmac-sha256.
+          RFC2136_TSIG_KEY=acme_example_org
+          RFC2136_PROPAGATION_TIMEOUT=1000
+          RFC2136_POLLING_INTERVAL=30
+          RFC2136_SEQUENCE_INTERVAL=30
+          RFC2136_DNS_TIMEOUT=1000
+          RFC2136_TTL=1
+          EOF
+        ";
+      };
+    };
+    type = types.attrsOf (types.submodule ({name, config, ...}: {
+      options = {
+        gpg = lib.mkOption {
+          type = types.path;
+          default =
```

Comment:
Perhaps this will cause the manual to depend on the configuration? It might be necessary to use `defaultText` to break this dependency.